projects/plugins/jetpack/_inc/client/main.jsx

```diff
@@ -198,6 +199,75 @@ class Main extends React.Component {
 	}
 
 	renderMainContent = route => {
+		/**
+		 * There are two conditions (groups of conditions, really) where we would want to
+		 * show the partner connection screen.
+		 *
+		 * 1. The site is not yet connected to WPCOM, but has the jetpack_partner_coupon
+		 * option set in the database (this.props.partnerCoupon in redux). This is likely a
+		 * partner-user who has just arrived here from a CTA within a partner's dashboard
+		 * or other ecosystem.
+		 *
+		 * 2. The site is already connected to WPCOM, but the jetpack_partner_coupon option
+		 * is still set in the database. This means the user connected their site, but never
+		 * redeemed the coupon. If this is the case, we don't want to override the dashboard
+		 * or at a glance pages with the redemption screen. Instead, we'll catch a URL
+		 * parameter that JITMs will set (showCouponRedemption=true), and show the screen only
+		 * when the user came from a a JITM.
+		 */
+		if ( this.props.partnerCoupon ) {
+			const forceShow = new URLSearchParams( window.location.search ).get( 'showCouponRedemption' );
+			const partnerCoupon = this.props.partnerCoupon;
+
+			if ( ! this.props.isSiteConnected || forceShow ) {
+				return (
+					<ConnectScreen
+						apiNonce={ this.props.apiNonce }
+						registrationNonce={ this.props.registrationNonce }
+						apiRoot={ this.props.apiRoot }
+						images={ [ '/images/connect-right-partner.png' ] }
```

Comment:
Three comments about this image (`/images/connect-right-partner.png`):

1. The size of it looks like we could compress it a bit more and maybe reduce the size by 60-80% through e.g. tinypng
2. Design will most likely comment that the size of the image isn't very "retina", so perhaps we should export it in a 2x size from Figma? :)
3. The image itself looks a lot like the product description backup image but with some fading green effect (`projects/plugins/jetpack/images/products/illustration-backup.png`). I think the intention from design here is to show a product relevant illustration

Reply:
Thanks, good catch! I think it's better to just use the existing image, so I've updated the patch to do so.

Reply:
I'm not sure if this is a new issue but while testing the update I noticed that there is a gap lacking background-color because we use `float: right;` and the image isn't "slim" enough for it to fill up all vertical space

![Screenshot 2021-11-22 at 21 40 51](https://user-images.githubusercontent.com/3846700/142932220-e539aac6-f79e-4949-bcba-37ed40246f48.png)